### ISSUE TITLE
Unexpected WRN_NullReferenceReturn warning for an async method producing non-generic Task

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29957]
```
        // PROTOTYPE(NullableReferenceTypes): Should not report WRN_NullReferenceReturn for F0.
        [WorkItem(23275, "https://github.com/dotnet/roslyn/issues/23275")]
        [Fact]
        public void AsyncTaskMethodReturningNull()
        {
            var source =
@"#pragma warning disable 1998
using System.Threading.Tasks;
class C
{
    static async Task F0() { return null; }
}";
            var comp = CreateCompilationWithMscorlib46(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            comp.VerifyDiagnostics(
                // (5,30): error CS1997: Since 'C.F0()' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task<T>'?
                //     static async Task F0() { return null; }
                Diagnostic(ErrorCode.ERR_TaskRetNoObjectRequired, "return").WithArguments("C.F0()").WithLocation(5, 30),
                // (5,37): warning CS8603: Possible null reference return.
                //     static async Task F0() { return null; }
                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "null").WithLocation(5, 37));
        }
```

**Comment by jcouv** [comment]
Need to confirm whether there is still an issue. Pushing out to Compiler.Next.
